src/promptflow/promptflow/_sdk/_submitter/run_submitter.py

```diff
@@ -39,6 +39,28 @@ def submit(self, run: Run, stream=False, **kwargs):
         self._run_bulk(run=run, stream=stream, **kwargs)
         return self.run_operations.get(name=run.name)
 
+    def resume(self, resume_from: str, **kwargs):
+        resume_from_run = self._ensure_required_run(resume_from)
+        run = resume_from_run._copy(resume_from=resume_from_run.name, **kwargs)
+        self._run_bulk(run=run, **kwargs)
+        return self.run_operations.get(name=run.name)
+
+    def _ensure_required_run(self, required_run: Union[str, Run]):
+        """Ensure the required run is completed and return the run object."""
+        if isinstance(required_run, str):
+            required_run = self.run_operations.get(name=required_run)
+        elif not isinstance(required_run, Run):
+            error = TypeError(f"Referenced run must be a Run instance, got {type(required_run)}")
+            raise UserErrorException(message=str(error), error=error)
+        else:
+            # get the run again to make sure it's status is latest
+            required_run = self.run_operations.get(name=required_run.name)
+        if required_run.status != Status.Completed.value:
+            error = ValueError(f"Referenced run {required_run.name} is not completed, got status {required_run.status}")
+            raise UserErrorException(message=str(error), error=error)
+        required_run.outputs = self.run_operations._get_outputs(required_run)
```

Comment:
why this?

Reply:
(I don't have enough context about this lol